SettingsList.py

```diff
@@ -2108,22 +2108,37 @@ def __init__(self, name, gui_text, min, max, default, step=1,
             'randomize_key': 'randomize_settings',
         },
     ),
-    Checkbutton(
+    Combobox(
         name           = 'shuffle_song_items',
-        gui_text       = 'Shuffle Songs with Items',
+        gui_text       = 'Shuffle Songs',
+        default        = 'song',
+        choices        = {
+            'song':    'Song Locations',
+            'dungeon': 'Dungeon Rewards',
+            'any':     'Anywhere',
+            },
         gui_tooltip    = '''\
-            Enabling this shuffles the songs into the rest of the
-            item pool.
+            This restricts where song items can appear.
+
+            'Song Locations': Song will only appear at vanilla song
+            locations. In Multiworld, songs will only appear in their
+            own world.
```

Comment:
I feel like this wording might be confused as songs being at their literal vanilla locations.